### DISCUSSION TITLE
Does Xandikos syncs the calendar colours?

### DISCUSSION THREAD

**Discussion by lamyergeier** [jelmer/xandikos#184] [Q&A] [answered]
Does Xandikos sync's the calendar colours? If yes, please tell me. I tried to set color on calendar from Xandikos server on Thunderbird both running on my MacOS. I use DAVx to sync it to my Android and use Simple Calendar. But the colors set on Thunderbird for each calendars are not reflected on Android!

**Comment by jelmer** [comment, marked as answer]
Xandikos supports storing the "calendar-color" property that Apple has defined as an extension to CalDAV, but which is not part of the official CalDAV standard. Some other clients (DAVx, Apple and Inf-It) have added support for this extension as well and can write and read it.

Not all clients support this extension, and that includes Thunderbird IIRC.

**Reply by lamyergeier** [reply]
I don't use any Apple specific tools. I only use open source tools. Both clients Thunderbird and Simple Mobile Calendar, DAVx5 are open source.

May I suggest that it would be nice if we could observe what Radicale does:

It creates a json file called as .Radicale.props inside each calendar (here, calendar name is Studies):

{"C:supported-calendar-component-set": "VEVENT", "D:displayname": "Studies", "ICAL:calendar-color": "#ffababff", "tag": "VCALENDAR"}

So, Radicale explicitly defines the color for each calendar. I have used DAVx with Radicale and the colours remain consistently. Currently all calendars have same colour on my Android and when I try to change those, after syncing with DAVx they become the same default colour (this was not the case when I used Radicale). I am guessing this issue is because Xadikos don't store the colour changed by clients using DAVx.

Is it possible to add this feature to Xandikos?

**Reply by jelmer** [reply]
Xandikos stores custom properties in a .xandikos file in the calendar directory.

Radicale is also storing the Apple property: "ICAL" in the example you paste above is short for http://apple.com/ns/ical/. See https://github.com/Kozea/Radicale/blob/master/radicale/xmlutils.py#L48. 

I know that DAV5x usest the Apple color property, and that's why that works for you when you use DAV5x with Radicale (and DAV5x will work too with Xandikos). I'm pretty sure Thunderbird doesn't support storing the calendar color in a WebDAV property (not the Apple property, nor anything else), it just only stores the color locally. There is no feature Xandikos or Radicale can add here, it's something that needs to be added in Thunderbird.

**Reply by lamyergeier** [reply]
You said: 

> Xandikos stores custom properties in a .xandikos file in the calendar directory.

But I don't find the color of calendar stored in this file. Where is it stored on the server for each calendar?

**Reply by jelmer** [reply]
If a client sets the color property (again, Thunderbird does not), Xandikos will store it in ``.xandikos`` in the calendar directory on the server. This is an ini-style file, the line will look something like ``color = #aabbcc``

If the line is not present, then no client has set the color.

**Reply by lamyergeier** [reply]
I am syncing the colour set by DAVx from Android.

Does Xandikos support colour from DAVx?

**Reply by jelmer** [reply]
Yes, see mention above - DAVx uses the Apple property.

**Reply by jelmer** [reply]
Actually, I'm not sure if DAVx can actually set the color - I don't see an option for it. It may just be able to read it.
